src/UnitConversions.hs

```diff
@@ -72,16 +75,39 @@ convertTemperature u s = unpack $ foldl replaceTemperature (pack s) (fmap packTe
         convertReplacement = fmap $ show . toTempUnit u
 
 findReplacements :: String -> [(String, Temperature)]
-findReplacements = map parseRegexResult . findTemperatures
-  where parseRegexResult l = (l!!0, Temperature (read (l!!1)) (parseTempUnit (l!!2)))
+findReplacements = map fromJust . filter isJust . map parseRegexResult . findTemperatures
+  where parseRegexResult r = to3Tuple r >>= parseTemperature
+
+to3Tuple :: [a] -> Maybe (a, a, a)
+to3Tuple [a,b,c] = return (a,b,c)
+to3Tuple _ = fail $ "can't convert non 3-element list to 3-Tuple"
+
+parseTemperature :: (String,String,String) -> Maybe (String, Temperature)
+parseTemperature (s,v,u) = case isJust maybeValue && isJust maybeUnit of
+  True -> return (s, Temperature (fromJust maybeValue) (fromJust maybeUnit))
```

Comment:
`fromJust` is another problematic function... If you're not familiar with the pitfalls of such (I think you mentioned you're somewhat new to Haskell?) I'm happy to discuss via email.

See also: 
 - https://wiki.haskell.org/Partial_functions
 - https://wiki.haskell.org/List_of_partial_functions

Reply:
I think I now got rid of all partial functions